### PR TITLE
Feature/add api logic for attach annex

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes changes related to:

- return id of the annex when attached to binding contract
- add command and handler for attachment of annex
- add endpoint for attachment of annex
- cleanup of existing contracts tests to stick to convention (do not include status_code in method descriptions)

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Api/AttachAnnexToBindingContract/AttachAnnexToBindingContractEndpoint.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using AttachAnnexToBindingContract;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+internal static class AttachAnnexToBindingContractEndpoint
+{
+    internal static void MapAttachAnnexToBindingContract(this IEndpointRouteBuilder app) => app.MapPost(
+            ContractsApiPaths.AttachAnnex, async (
+                Guid id,
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Api/AttachAnnexToBindingContract/AttachAnnexToBindingContractEndpoint.cs:13]: Could we change to Guid bindingContractId?

**Review by kamilbaczek — APPROVED** [review]
I suggested some small improvemnts. Overall looks very good!

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
please check comments

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs [hunk 2]
```diff
@@ -44,15 +44,19 @@ public sealed class BindingContract : Entity
...
 
         BusinessRuleValidator.Validate(
             new AnnexCanOnlyStartDuringBindingContractPeriodRule(ExpiringAt, validFrom));
 
-        AttachedAnnexes.Add(Annex.Attach(Id, validFrom));
+        var annex = Annex.Attach(Id, validFrom);
+
+        AttachedAnnexes.Add(annex);
+
+        return annex.Id;
```
[the remark this answers is not in the document]
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/BindingContract.cs:59]: It is already raised in Annex :)

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**